File: test_base/c/code/int_alloc_n_write_m.c
#include <stdlib.h>
#include <stdio.h>

int main(int argc, char const *argv[])
{
    if (argc < 2) 
    {
        return 1;
    }

    int n;

    sscanf(argv[1], "%d", &n);

    printf("%d\n", n);
    
    if (n < 0)
    {
        return 2;
    }

    int *p = malloc(n * sizeof(int));

    printf("alloc done\n");

    if (argc == 3)
    {
        int m;

        sscanf(argv[2], "%d", &m);

        if (m < 0 || m > n)
        {
            return 3;
        }

        printf("%d\n", m);

        for (int i = 0 ; i < m ; i++)
        {
            p[i] = 8;
        }

        printf("write done\n");
    }
    
    free(p);
    
    return 0;
}

File: test_base/python/code/int_alloc_n_write_m.py
import sys
import array

if __name__ == "__main__":
    if len(sys.argv) < 2:
        exit(1)
    
    n = int(sys.argv[1])

    if n < 0:
        exit(2)
    
    #print(n)

    p = array.array('i', (0 for i in range(0, n)))

    #print("alloc done")

    if len(sys.argv) == 3:
        m = int(sys.argv[2])

        if m < 0 or m > n:
            exit(3)
        
        #print(m)
        
        for i in range(m):
            p[i] = 8
        
        #print("write done")
    
    exit(0)
